src/components/SubjectPhasePicker/SubjectPhasePicker.tsx

```diff
@@ -243,15 +247,83 @@ const SubjectPhasePicker: FC<SubjectPhasePickerData> = ({
     );
   };
 
+  const ButtonFocusUnderline = styled(Svg)<{ $color: OakColorName }>`
+    color: ${(props) => props.$color};
+  `;
+
+  /**
+   * ! - TODO LIST
+   * TODO: Refactor to break down into smaller components
+   */
+
+  const viewButtonRef = useRef<HTMLButtonElement>(null);
+  const depsRef = useRef(
+    selectedSubject &&
+      selectedPhase &&
+      path &&
+      path.startsWith("/teachers/curriculum/"),
+  );
+
+  useEffect(() => {
```

Comment:
Suggest refactoring for clarity, and using states rather than setting colors directly to decouple logic & style. Here's the re-factor:
```
useEffect(() => {
  // Initial values
  let newDisplayNewBorders = false;
  let newPhaseEnabled = true;
  let newSubjectEnabled = true;

  if (showSubjects) {
    // Subjects showing, show borders and disable phase
    newDisplayNewBorders = true;
    newPhaseEnabled = false;
  } else if (showPhases) {
    // Phases showing, show borders and disable subject
    newDisplayNewBorders = true;
    newSubjectEnabled = false;
  } else if (selectedSubject && selectedPhase) {
    // Subject and phase selected, focus on "View" button
    viewButtonRef.current?.focus();
  }

  // Set states
  setDisplayNewBorders(newDisplayNewBorders);
  setPhaseEnabled(newPhaseEnabled);
  setSubjectBackground(newSubjectEnabled);
}, [selectedSubject, selectedPhase, showPhases, showSubjects]);
```

You would need to track `phaseEnabled` and `subjectEnabled` instead of `phaseBackground` and `subjectBackground`.

The actual colors can be handled in the HTML portion of the code, ie in phases:
```
<Box
  $position="relative"
  data-testid="subjectPhasePicker"
  $zIndex={"mobileFilters"}
  $background={phaseEnabled ? "white" : "grey1"}
  $maxWidth={960}
>
```

Reply:
A lot of excess logic, cleaned up 😅